go/vt/tabletserver/query_rules.go

```diff
@@ -689,6 +689,13 @@ func getstring(val interface{}) (sv string, status int) {
 //-----------------------------------------------
 // Support functions for JSON
 
+func MapStrOperator(strop string) (op Operator, err error) {
+	if op, ok := opmap[strop]; ok {
+		return op, nil
+	}
+	return QR_NOOP, NewTabletError(ErrFail, "Cannot map string %s to a valid Operator", strop)
```

Comment:
Why is this error different from the other one? You could instead move that error here, and return it as is in BuildQueryRule.